libunwind/src/UnwindRegistersSave.S

```diff
@@ -824,6 +824,64 @@ DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
 //
   .p2align 2
 DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
+#ifdef __CHERI_PURE_CAPABILITY__
+  stp    c0, c1,  [c0, #0x000]
+  stp    c2, c3,  [c0, #0x020]
+  stp    c4, c5,  [c0, #0x040]
+  stp    c6, c7,  [c0, #0x060]
+  stp    c8, c9,  [c0, #0x080]
+  stp    c10,c11, [c0, #0x0A0]
+  stp    c12,c13, [c0, #0x0C0]
+  stp    c14,c15, [c0, #0x0E0]
+  stp    c16,c17, [c0, #0x100]
+  stp    c18,c19, [c0, #0x120]
+  stp    c20,c21, [c0, #0x140]
+  stp    c22,c23, [c0, #0x160]
+  stp    c24,c25, [c0, #0x180]
+  stp    c26,c27, [c0, #0x1A0]
+  stp    c28,c29, [c0, #0x1C0]
+  str    c30,     [c0, #0x1E0]
+  mov    c1,csp
+  str    c1,      [c0, #0x1F0]
+  str    c30,     [c0, #0x200]    // store return address as pc
+  // skip cpsr
+  // XXX use str instead of stp below because
+  // of immediate range restrictions
```

Comment:
https://git.morello-project.org/morello/llvm-project/-/merge_requests/40 uses stp.

Reply:
Yes, it has the optimization Jess wanted of adjusting the base pointer. I was going to wait to pull that in, but I might as well merge it now.